source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -651,9 +652,17 @@ public Task<int> ExecuteNonQueryAsync(string query,  CommandParameterValues? arg
         public int ExecuteNonQuery(PreparedCommand preparedCommand)
         {
             using var mutex = DeadlockAwareLock.Lock();
-            using var command = CreateCommand(preparedCommand);
-            var result =  command.ExecuteNonQuery();
+            var retryPolicy = GetRetryPolicy(preparedCommand.Operation).LoggingRetries(nameof(ExecuteNonQuery));
+            var result = retryPolicy.ExecuteAction(() =>
+            {
+                if (connection?.State is ConnectionState.Closed)
```

Comment:
Is this enough to determine that we need to replay the transaction?

Reply:
I am not sure to be honest. Should it be using the IsTransient check? Is that possible here?

Reply:
The "replay" state should be determined when detecting the transient error